readme.md

```diff
@@ -131,3 +137,24 @@ Se considera la instalación de algunos periféricos sobre una ventana de dos ho
 | Precondición      | La alimentación del sistema está conectada y el sistema está activo.                                                                                                                                                                                                                                                                                                        |
 | Flujo básico      | El usuario ingresa la contraseña correcta mediante el teclado matricial, el buzzer 2 suena dando dos pitidos con un silencio en el medio y el sistema entra en modo inactivo                                                                                                                                                                                                |
 | Flujo alternativo | El usuario no ingresa la contraseña correcta o demora más de treinta segundos en completar el ingreso de la contraseña. En ese caso el sistema el sistema da una segunda oportunidad para ingresar la contraseña correcta pero el buzzer 2 debe emitir pitidos intermitentes. Si se ingresa una contraseña incorrecta o se agota el tiempo de espera, debe sonar la bocina. |
+
+
+## Informe de avance
+- Presentación: 31/08/2024
+
+El estado del proyecto en la fecha de presentacón se presenta en la siguiente tabla.
+![arm_book_curso_2024_TP_final_tabla_informe_avance](https://github.com/user-attachments/assets/0e5fc095-4fea-43cc-b81b-8449981e8105)
+
+### Observaciones
+- Se modifica el requisito 5.2. 
+  - Requisito original: "No debe sonar la bocina si el estado de alarma no está activo o si el mismo está activo pero se ingresa la contraseña correcta para desactivar el sistema"
+  - Requisito modificado: "No debe sonar la bocina si el estado de alarma no es 'activo'"
+  - Motivo: Un cambio de opinión sobre el diseño. La bocina sólo debe sonar si el estado del sistema es 'activo'. Cuando se introduce la contraseña correcta el estado debe pasar a 'inactivo'.
+
+- Se corrige error de redacción en requisito 6.2
+  - Requisito original: "No debe realizar disparar ninguna acción si el pulsador no es presionado"
+  - Requisito modificado: "No debe disparar ninguna acción si el pulsador no es presionado"
+ 
+- Sobre el requisito 7.1: A la fecha de presentación aún no se pudo implementar la interfaz de usuario mediante smartphone debido a inconvenientes con el [firmware](https://docs.espressif.com/projects/esp-at/en/latest/esp32/Get_Started/What_is_ESP-AT.html) para recibir comandos AT en la placa nodemcu ESP32. En los días siguientes se intentará lograr la conectividad propuesta, en caso contrario se modificará para implementar la interfaz de usuario mediante terminal bluetooth en smartphone.
```

Comment:
defina "inconvenientes con el firmware".
Hasta donde yo sé, el firmware goza de buena salud, el único inconveniente sería el usuario no entendiendo la documentación del mismo o alguna otra cosa de índole similar

Reply:
En cierto punto de la guía de espressif había que instalar el entorno del proyecto esp-at mediante la instalación de unos paquetes de Python. Luego había que ejecutar un comando de python para abrir un menú de configuración y configurar parámetros de la placa esp32. En ese paso la consola arrojaba un error, lo que no permitía construir el proyecto (ejecutar el comando build). Luego en lugar de descargar el firmware opté por [realizar una compilación local](https://docs.espressif.com/projects/esp-at/en/latest/esp32/Compile_and_Develop/How_to_clone_project_and_compile_it.html#compile-esp-at-project-locally) en Windows y no pude (ya ni recuerdo qué error tuve de tanto que di vueltas con el tema). Pero cuestión que pude solucionar compilando el proyecto localmente en Linux (incluso fue más sencillo). Sin embargo también tuve unos traspies en Linux por ejemplo: al construir el proyecto la herramienta quería ejecutar automáticamente ciertos comandos de python y arrojaba error teniendo instalado python; lo solucioné instalando el paquete 'python-is-python3'.

Finalmente, anoche pude realizar correctamente el flash del firmware del proyecto esp-at en la placa esp32 y todo funciona correctamente. Escribí en el puerto serie mediante PuTTY el comando 'AT' y recibí la respuesta 'OK'. Luego seguí los pasos de la sección 11.2.2 del libro del curso y todo concluyó bien. Pude ver en mi smartphone la página HTML con el mensaje 'Hello!' que aparece en el libro. Asi que el requisito 7.1 pasa de color rojo a amarillo.

Reply:
Agrego lo siguiente porque es un detalle que le puede servir a alguien más:
Cuando tenía el firware cargado y listo para probar, al abrir un puerto serie en PuTTY, escribir el comando 'AT' y presionar la tecla 'enter', el cursor volvía al principio de la línea (como que no envíaba el comando). En general al presionar enter se envía el retorno de carro y salto de línea pero no sé porque la herramienta no lo hacía. [Descubrí que](https://z49x2vmq.github.io/2017/11/12/putty-cr-lf-en/), como alternativa al enter, se pueden usar los atajos de teclado Ctrl+M y Ctrl+J para CR y LF respectivamente. Entonces escribí 'AT', luego Ctrl+M, luego Ctrl+J y ahí sí, finalmente, recibía el bendito 'OK'. Luego procedí de manera similar para el resto de los comandos AT.

Reply:
No, no se envía ni se deja de enviar comandos, PuTTY es un software de TTY; una TTY es un simple teclado -> serie + serie -> pantalla; no tiene idea de qué es "un comando" (a menos que lo configures para hacer cosas más complejas). La tecla ENTER envía un CR, para que envíe además LF hay que configurarlo para que lo haga. Simplemente buscá en la configuración del prog de emulación de terminal (sea PuTTY, minicom, o el que se te ocurra) y configuralo para que envíe LF luego de CR.
No son atajos de teclado, son formas de enviar CR y LF por separado. Ambos Ctrl corresponden cada uno al código ASCII de CR y LF respectivamente; si mandás Ctrl-G hace un beep, porque ése caracter es BEL... etc. Hay que mirar el diálogo en el puerto serie con un sniffer, suele ser más preciso que adivinar.

Reply:
Gracias por las aclaraciones. No me percaté de que al apretar enter sólo hacía un CR, es obvio porque literalmente el cursor de color verde volvía al principio. Pero como no había usado mucho PuTTY anteriormente, había interpretado que algo andaba mal pero en realidad faltó esa configuración de la tecla enter. Ahora encontré esa opción para que envíe LF junto con CR.